# bioRxiv_05052022_Hdac_dual_roles


Current:
Jeff Jiajing Zhou, Jin Sun Cho, Han Han, Ira L Blitz, Wenqi Wang, Ken WY Cho (2023) Histone deacetylase 1 maintains lineage integrity through histone acetylome refinement during early embryogenesis eLife 12:e79380
https://doi.org/10.7554/eLife.79380


Former:
Zhou, J. J., Cho, J. S., Han, H., Blitz, I. L., Wang, W., & Cho, K. W. Y. (2022). Histone deacetylase 1 maintains lineage integrity through histone acetylome refinement during early embryogenesis. BioRxiv, 490762. https://doi.org/10.1101/2022.05.05.490762




Majory data type generated in this paper: ChIP-seq, RNA-seq

Data contents (Xenopus tropicalis embryos):
1. Hdac1 ChIP-seq time course (st8, st9, and st10.5) in WT whole embryos
2. Hdac2 ChIP-seq (st9 and st10.5) in WT whole embryos
3. Hdac1 ChIP-seq in st9 embryos with a-amanitin injection
4. Hdac1 ChIP-seq in st9 and st10.5 embryos under DMSO/TSA treated conditions
5. Sox3 ChIP-seq in st9 WT whole embryos
6. H3K18ac ChIP-seq in st9 WT whole embryos
7. pan-H3Kac ChIP-seq in st10.5 gastrula ectoderm/endoderm under DMSO/TSA treated conditions
8. RNA-seq in st10.5 gastrula ectoderm/endoderm under DMSO/TSA treated conditions
9. RNA-seq in st10.5 gastrula ectoderm/endoderm under H2O/VPA treated conditions



Scripts presented here are mainly specific analyses to generate figures, common bioinformatic tools used upstream of the analyses include but are not limited to:
1. STAR
2. Samtools
3. Bedtools
4. Deeptools
5. RSEM
6. Homer
7. Bioconda: IDR
